[Fleet] Agent status alerts for monitoring

karenzone opened the pull request:
## Summary
Updates monitoring docs to encourage use of new [Elastic Agent out-of-the-box alert rules](https://www.elastic.co/docs/reference/fleet/alert-templates#ea-alert-rules). 

Closes: https://github.com/elastic/ingest-docs/issues/1873
Closes: #5315 

## Generative AI disclosure
<!--
To help us ensure compliance with the Elastic open source and documentation guidelines, please answer the following:
-->
1. Did you use a generative AI (GenAI) tool to assist in creating this contribution?
- [ ] Yes  
- [x] No

karenzone commented on reference/fleet/monitor-elastic-agent.md:
```diff
@@ -241,11 +241,20 @@ To do so, follow the steps in [Remote {{es}} output](/reference/fleet/remote-ela
+:::{admonition} Built-in alert rules for {{agent}}
+:applies_to: stack: ga 9.2.0
+
+As of 9.2.0, {{agent}} provides out-of-the-box alert rules to help you simplify monitoring the health of agents.
```
```suggestion
As of 9.2.0, {{agent}} provides out-of-the-box alert rules to simplify monitoring the health of agents.
```

nchaulet commented:
> @nchaulet @MichelLosier I was wondering if you had a recording on how to use the new alerts?


I recored this quick one for Karen sometime ago if it can help https://github.com/user-attachments/assets/2490f16d-4b1c-4d4b-9e95-3c745d0d0e6a

nimarezainia commented:
> > @nchaulet @MichelLosier I was wondering if you had a recording on how to use the new alerts?
> 
> I recored this quick one for Karen sometime ago if it can help https://github.com/user-attachments/assets/2490f16d-4b1c-4d4b-9e95-3c745d0d0e6a

thank you. I think we need to dive a bit deeper into the agent unhealthy rule template in the docs. I'll play with that a bit more.

karenzone commented on reference/fleet/alert-templates.md:
```diff
@@ -29,13 +29,16 @@ You can find these rules in **Stack Management** > **Alerts and Insights** > **R
+| [Elastic Agent] Excessive memory usage|  Checks if {{agent}} or any of its processes have a high memory usage or memory usage that is trending up. This could signal a memory leak in an application and warrant further investigation.<br>- Condition: Alert on `system.process.memory.rss.pct` more than 50%<br>- Default: Disabled |
+| [Elastic Agent] Excessive restarts| Checks for excessive restarts on a host. Some restarts can have a business impact, and getting alerts for them can enable timely mitigation.<br>- Condition: Alert on 11 or more restarts in a 5-minute window<br>- Default: Disabled |
+| [Elastic Agent] High pipeline queue | Checks percentage of pipeline queue. Rows are distinguished by agent ID and component ID. <br> - Condition: Alert on max of `beat.stats.libbeat.pipeline.queue.filled.pct` exceeding 90%  <br>- Default: Disabled|
+| [Elastic Agent] Offline status | Desc <br> - Condition:  <br>- Default: Disabled|
```
Add details

karenzone commented on reference/fleet/alert-templates.md:
```diff
@@ -29,13 +29,16 @@ You can find these rules in **Stack Management** > **Alerts and Insights** > **R
+| [Elastic Agent] Output errors | Checks errors per minute from an agent component. Rows are distinguished by agent ID and component ID. <br> - Condition: Alert on 6 or more errors per minute  <br>- Default: Disabled|
+| [Elastic Agent] Unenrolled status | removed from Fleet, API keys revoked, no data flow, re-enrollment required <br> - Condition:  <br>- Default: Disabled|
+| [Elastic Agent] Unhealthy status | Checks agent status. An `unhealthy` status can indicate errors or degraded functionality of the agent. <br> - Condition: Alert on `unhealthy` status <br>- Default: Disabled|
+| [Elastic Agent] Uninstalled status | Desc <br> - Condition:  <br>- Default: Disabled|
```
Add details

karenzone commented:
@nimarezainia, I moved OOB alerts to a more appropriate place in the TOC structure. As we discussed, they were there all long, but not in the best location.  Thank you for working through that with me. 

@MichelLosier, I added the new OOB alerts (#5315). As always, I appreciate your help. 

@vishaangelova, I used the latest `applies_to` functionality--specifically switches and ranges--to present the new and improved alerting content while preserving docs for 9.0-9.1 functionality. Thanks to you and the team for expanding versioning functionality to make it easier to communicate versioning status to our users.

MichelLosier commented on reference/fleet/alert-templates.md:
```diff
@@ -29,13 +29,16 @@ You can find these rules in **Stack Management** > **Alerts and Insights** > **R
+| [Elastic Agent] Excessive memory usage|  Checks if {{agent}} or any of its processes have a high memory usage or memory usage that is trending up. This could signal a memory leak in an application and warrant further investigation.<br>- Condition: Alert on `system.process.memory.rss.pct` more than 50%<br>- Default: Disabled |
+| [Elastic Agent] Excessive restarts| Checks for excessive restarts on a host. Some restarts can have a business impact, and getting alerts for them can enable timely mitigation.<br>- Condition: Alert on 11 or more restarts in a 5-minute window<br>- Default: Disabled |
+| [Elastic Agent] High pipeline queue | Checks percentage of pipeline queue. Rows are distinguished by agent ID and component ID. <br> - Condition: Alert on max of `beat.stats.libbeat.pipeline.queue.filled.pct` exceeding 90%  <br>- Default: Disabled|
+| [Elastic Agent] Offline status <br> {applies_to}`stack: ga 9.3+`| Checks for any agents that are offline. <br> - Condition: Alert when agent has been offline for longer than the time set in `inactivity timeout` <br>- Default: Disabled|q
```
I don't think we need to flag these rules as 9.3+, although they came later after initial release, these additions were purely changes to the integration package and are available to 9.2 as well.

MichelLosier reviewed (changes requested):
This looks great! Just a couple small items and should be good ✅

vishaangelova commented on reference/fleet/alert-templates.md:
On Serverless, I entered "Stack Management" in the search field, and when I opened it, I expected to have to click an "Alerts and Insights" menu to get to "Rules".

Instead, "Alerts and insights" is a section on the landing page, but the "Rules" menu is not visible under it. The “Rules" menu appears in the left side nav in the “Alerts and Insights” section (the latter is not clickable)… So this may be a bit confusing.

On ECH with a 9.3 deployment, the experience was similar, but maybe it was less confusing because there was no "Alerts and Insights" section on the landing page, there was just the left side nav menu section.

vishaangelova commented on reference/fleet/alert-templates.md:
```diff
@@ -29,13 +29,16 @@ You can find these rules in **Stack Management** > **Alerts and Insights** > **R
+| [Elastic Agent] Excessive memory usage|  Checks if {{agent}} or any of its processes have a high memory usage or memory usage that is trending up. This could signal a memory leak in an application and warrant further investigation.<br>- Condition: Alert on `system.process.memory.rss.pct` more than 50%<br>- Default: Disabled |
+| [Elastic Agent] Excessive restarts| Checks for excessive restarts on a host. Some restarts can have a business impact, and getting alerts for them can enable timely mitigation.<br>- Condition: Alert on 11 or more restarts in a 5-minute window<br>- Default: Disabled |
+| [Elastic Agent] High pipeline queue | Checks percentage of pipeline queue. Rows are distinguished by agent ID and component ID. <br> - Condition: Alert on max of `beat.stats.libbeat.pipeline.queue.filled.pct` exceeding 90%  <br>- Default: Disabled|
+| [Elastic Agent] Offline status | Checks for any agents that are offline. <br> - Condition: Alert when agent has been offline for longer than the time set in `inactivity timeout` <br>- Default: Disabled|q
```
The table renders a third empty column (see the preview) because of the `q` here, so:

```suggestion
| [Elastic Agent] Offline status | Checks for any agents that are offline. <br> - Condition: Alert when agent has been offline for longer than the time set in `inactivity timeout` <br>- Default: Disabled|
```

karenzone commented on reference/fleet/alert-templates.md:
```diff
@@ -29,13 +29,16 @@ You can find these rules in **Stack Management** > **Alerts and Insights** > **R
 
 | Alert | Description |
```
We went through much discussion and several iterations on initial docs, and this is where we landed.  I'm always open to revisiting such things, but I don't want to hold this one up any longer.
